src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -227,6 +237,814 @@ class OMPSerialDirective(OMPRegionDirective, metaclass=abc.ABCMeta):
+            # Found a range index between dependencies which does not cover
+            # the full array range, which is not currently supported in
+            # PSyclone (due to OpenMP limitations).
+            return False
+        return True
+
+    def _compute_accesses(self, ref, preceding_nodes, task):
+        '''
+        Computes the set of accesses for a Reference or BinaryOperation
+        Node, based upon the preceding_nodes and containing task.
+
+        The returned result is either a set of Literals, or Literals
+        and BinaryOperations.
+
+
+        :param ref: The Reference or BinaryOperation node to compute \
+                    accesses for.
+        :type ref: Union[:py:class:`psyclone.psyir.nodes.Reference, \
+                   :py:class:`psyclone.psyir.nodes.BinaryOperation]
+        :param preceding_nodes: A list of nodes that precede the task in the \
+                                tree.
+        :type preceding_nodes: list of :py:class:`psyclone.psyir.nodes.Node`
+        :param task: The OMPTaskDirective node containing ref as a child.
+        :type task: :py:class:`psyclone.psyir.nodes.OMPTaskDirective`
+
+        :raises GenerationError: If the ref contains an unsupported \
+                                 BinaryOperation structure, such as a \
+                                 non-ADD/SUB/MUL operator. Check error \
+                                 message for more details.
+        :raises GenerationError: If preceding_nodes contains a Call node.
+        :raises GenerationError: If ref is a BinaryOperation and neither \
+                                 child of ref is a Literal or BinaryOperation.
+        :raises GenerationError: If there is a dependency between ref (a \
+                                 BinaryOperation) and a previously set \
+                                 constant.
+        :raises GenerationError: If there is a dependency between ref and a \
+                                 Loop variable that is not an ancestor of \
+                                 task.
+        :raises GenerationError: If preceding_nodes contains a dependent loop \
+                                 with a non-Literal step.
+
+        :returns: a list of the dependency values for the input ref.
+        :rtype: list of Union[:py:class:`psyclone.psyir.nodes.Literal`, \
+                :py:class:`psyclone.psyir.nodes.BinaryOperation`].
+        '''
+        if isinstance(ref, Reference):
+            symbol = ref.symbol
+        else:
+            # Get the symbol out of the Binop, and store some other
+            # important information
+            if isinstance(ref.children[0], Literal):
+                if ref.operator == BinaryOperation.Operator.ADD:
+                    symbol = ref.children[1].symbol
+                    binop_val = int(ref.children[0].value)
+                    num_entries = 2
+                else:
+                    raise GenerationError("Found a dependency index that is "
```

Comment:
I'm not sure here - I think this is more to avoid issues with OpenMP, rather than my choice to not support it. I could say this is an OpenMP limitation (as opposed to PSyclone) in the text? 

I do think maybe GenerationError nor NotImplementedError are the right Error to throw, this could almost just be a generic Exception since these errors are used entirely to tell `_compare_ref_binop` to `return False` (and the method is otherwise private).

Reply:
Ah. In the frontend we use `NotImplementedError` to indicate that something needs to be put into a CodeBlock. Given that this is a special case you could add your own subclass of PSycloneError (I think it is) and use that, e.g. UnresolvedDependencyError or something. That would make the code clearer.

Reply:
Done